File: Practice(Beginner)/C++/Greedy Puppy.cpp
// Source: https://www.codechef.com/problems/GDOG

#include <bits/stdc++.h>
using namespace std;

int main() {

    int t, n, k;
    cin >> t;
    while (t--){
        cin >> n >> k;
        int maxProfit = 0;
        for (int i = 1; i <= k; ++i) {
            maxProfit = max(maxProfit,n%i);
        }
        cout << maxProfit << endl;
    }
    return 0;
}

File: Practice(Beginner)/Python3/Greedy_Puppy.py
# Source: https://www.codechef.com/problems/GDOG

t = int(input())
while(t):
    n, k = map(int, input().split())
    maxProfit = 0;
    for i in range(1,k+1):
        maxProfit = max(maxProfit,n%i)
    print(maxProfit)
    t -= 1
